compiler/rustc_borrowck/src/polonius/mod.rs

```diff
@@ -35,5 +35,146 @@
 
 mod constraints;
 pub(crate) use constraints::*;
-
 pub(crate) mod legacy;
+
+use rustc_middle::mir::{Body, Location};
+use rustc_middle::ty::TyCtxt;
+use rustc_mir_dataflow::points::PointIndex;
+
+use crate::RegionInferenceContext;
+use crate::constraints::OutlivesConstraint;
+use crate::region_infer::values::LivenessValues;
+use crate::type_check::Locations;
+use crate::universal_regions::UniversalRegions;
+
+/// When using `-Zpolonius=next`, fills the given constraint set by:
+/// - converting NLL typeck constraints to be localized
+/// - encoding liveness constraints
+pub(crate) fn create_localized_constraints<'tcx>(
+    regioncx: &mut RegionInferenceContext<'tcx>,
+    tcx: TyCtxt<'tcx>,
+    body: &Body<'tcx>,
+    localized_outlives_constraints: &mut LocalizedOutlivesConstraintSet,
+) {
+    if !tcx.sess.opts.unstable_opts.polonius.is_next_enabled() {
```

Comment:
This flag is checked twice: once in the caller, and one here. One of them should probably be removed (I have no real preference which).